### PR TITLE
KAFKA-8942: Document RocksDB metrics

### PR DESCRIPTION
**Pull request by cadonna**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cadonna** [comment]
Call for review: @bbjeck @ableegoldman @mjsax 

This PR needs to be cherry-picked to 2.4.

**Comment by cadonna** [comment]
JDK 11/Scala 2.13 and JDK 8/Scala 2.11 failed and logs on Jenkins have been already removed.

In JDK 11/Scala 2.12 the following test failed:
```
org.apache.kafka.trogdor.coordinator.CoordinatorTest.testTaskRequestWithOldStartMsGetsUpdated
```

Retest this, please

File: docs/ops.html [hunk 1]
```diff
@@ -1979,6 +1979,108 @@ <h5><a id="kafka_streams_store_monitoring" href="#kafka_streams_store_monitoring
...
+      <td>kafka.streams:type=stream-state-metrics,client-id=([-.\w]+),task-id=([-.\w]+),[store-scope]-id=([-.\w]+)</td>
+    </tr>
+    <tr>
+      <td>block-cache-index-hit-ratio</td>
+      <td>The ratio of block cache hits for indexes relative to all lookups for indexes to the block cache.</td>
+      <td>kafka.streams:type=stream-state-metrics,client-id=([-.\w]+),task-id=([-.\w]+),[store-scope]-id=([-.\w]+)</td>
+    </tr>
+    <tr>
+      <td>block-cache-filter-hit-ratio</td>
+      <td>The ratio of block cache hits for filters relative to all lookups for filters to the block cache.</td>
```
Comment (guozhangwang) [inline docs/ops.html:2048]: nit: `for filters` -> `for bloom filters`?

File: docs/ops.html [hunk 2]
```diff
@@ -1979,6 +1979,108 @@ <h5><a id="kafka_streams_store_monitoring" href="#kafka_streams_store_monitoring
...
+      <td>kafka.streams:type=stream-state-metrics,client-id=([-.\w]+),task-id=([-.\w]+),[store-scope]-id=([-.\w]+)</td>
+    </tr>
+    <tr>
+      <td>bytes-written-compaction-rate</td>
+      <td>The average number of bytes written per second during compaction.</td>
+      <td>kafka.streams:type=stream-state-metrics,client-id=([-.\w]+),task-id=([-.\w]+),[store-scope]-id=([-.\w]+)</td>
+    </tr>
+    <tr>
+      <td>number-open-files</td>
+      <td>The number of currently open files.</td>
```
Comment (guozhangwang) [inline docs/ops.html:2073]: nit: current open?

**Review by guozhangwang — COMMENTED** [review]
Just minor comment, otherwise lgtm.

**Comment by guozhangwang** [comment]
Merged to trunk and 2.4.0 with a minor squashed commit to address my own comments.

File: docs/ops.html (hunk 2 above)
Comment (ableegoldman) [reply docs/ops.html:2073]: "currently open" sounds right to me 😄

File: docs/ops.html (hunk 1 above)
Comment (ableegoldman) [reply docs/ops.html:2048]: Based on the APIs in rocks it actually does seem like there could be other kinds of filters, maybe added later or possibly meant for you to plug in your own. But, AFAIK bloom filters are the only real possibility atm so probably better to just be explicit

Comment (guozhangwang) [reply docs/ops.html:2048]: I ended up with `filter blocks`, `index blocks` and `data blocks` in the description.

**Pull request closed without merging**